File: steps/daily_report_step.py
config = {
    "name": "DailyReport",
    "type": "cron",
    "cron": "0 9 * * *",  # Run daily at 9 AM
    "flows": ["reports"],
}
 
async def handler(context):
    context.logger.info("Running daily report")
    await context.emit({
        "topic": "report.generated",
        "data": {"date": datetime.now().isoformat()},
    })

File: steps/daily-report.step.js
const config = {
  name: 'DailyReport',
  type: 'cron',
  cron: '0 9 * * *', // Run daily at 9 AM
  flows: ['reports'],
}
 
const handler = async ({ emit, logger }) => {
  logger.info('Running daily report')
  await emit({
    topic: 'report.generated',
    data: { date: new Date().toISOString() },
  })
}
 
module.exports = { config, handler }